python_modules/dagster/dagster/daemon/sensor.py

```diff
@@ -497,6 +501,40 @@ def _is_under_min_interval(state, external_sensor, now):
     return elapsed < external_sensor.min_interval_seconds
 
 
+def _fetch_existing_runs(instance, external_sensor, run_requests):
+    run_keys = [run_request.run_key for run_request in run_requests if run_request.run_key]
+    existing_runs = {}
+
+    if run_keys and instance.supports_bucket_queries:
```

Comment:
could stick an initial `if not run_keys return {}` case here vs. checking it in both cases, meh